app/features/notes_generator/core.py

```diff
@@ -0,0 +1,69 @@
+import re
+from typing import Literal, Optional, Dict
+import logging
+from tools import (
+    extract_text_from_file,
+    extract_text_from_url,
+    generate_notes,
+    export_notes
+)
+
+def executor(input_type: Literal['text', 'file', 'url'],
+             input_content: Optional[str] = None,
+             file_path: Optional[str] = None,
+             focus_topic: Optional[str] = None,
+             output_structure: Literal['bullet', 'paragraph', 'table'] = 'bullet',
+             export_format: Literal['txt', 'docx', 'pdf'] = 'txt') -> Dict[str, str]:
+    """
+    Main executor function for Notes Generator Tool.
+    Parameters:
+            input_type (str): Type of input - 'text', 'file', or 'url'.
+            input_content (str): Direct text input (if input_type='text').
+            file_path (str): Path to uploaded file (if input_type='file').
+            focus_topic (str): Topic or focus for notes generation.
+            output_structure (str): Format for notes - bullet, paragraph, or table.
+            export_format (str): Export format - txt, docx, or pdf.
+        Returns:
+            dict: Status and file path of the exported notes.
+            """
+    try:
+        logging.info("Notes generation started...")
+
+        # Validate that at least one of input_content or file_path is provided
+        if not (input_content or file_path):
+            raise ValueError("Either 'input_content' or 'file_path' must be provided.")
+        # Extract content based on input type
+        if input_type == 'text' and input_content:
+            content = input_content
+        elif input_type == 'file' and file_path:
+            # This is a check to validate file type before proceeding
+            if not file_path.endswith(('.txt', '.docx', '.pdf', '.csv')):
```

Comment:
These are not all the current file types that our backend supports. You can check our `document_loaders.py` for more insights on that.